tests/unit_tests/lcls_tools/common/image_processing/test_image_processing.py

```diff
@@ -1,88 +1,63 @@
-        images = []
-        while len(images) < 10:
-            images.append(self.MI.image)
-
-        ave_image = ip.average_image(images)
-        self.assertEqual(np.array_equal(ave_image, self.MI.image), True)
-
-    def test_shape_image(self):
-        """Test that we can reshape our ndarray"""
-        self.assertEqual(self.MI.image.shape, (1024, 1392))
-        image = ip.shape_image(self.MI.image, 16, 89088)
-        self.assertEqual(image.shape, (89088, 16))
-
-    def test_x_projection(self):
-        """Test we get expected value for x projection"""
-        x_proj = ip.x_projection(self.MI.image)
-        self.assertEqual(x_proj.sum(), 9792279)
-        self.assertEqual(int(x_proj.mean()), 7034)
-        self.assertEqual(int(x_proj.std()), 10005)
-
-    def test_y_projection(self):
-        """Test that we get expected value for y projection"""
-        y_proj = ip.y_projection(self.MI.image)
-        self.assertEqual(y_proj.sum(), 9623943)
-        self.assertEqual(int(y_proj.mean()), 9398)
-        self.assertEqual(int(y_proj.std()), 10398)
-
-    def test_gauss_func(self):
-        """Test we get correct value for a gaussian evaluation"""
-        ans = ip.gauss_func(1.0, 2.0, 3.0, 4.0)
-        self.assertEqual(round(ans, 2), 1.76)
-
-    def test_gauss_fit(self):
-        """Test that we get the correct gaussian fit parameters"""
-        x_proj = ip.x_projection(self.MI.image)
-        y_proj = ip.y_projection(self.MI.image)
-        _, a_x, x0_x, sigma_x = ip.gauss_fit(x_proj)
-        _, a_y, y0_y, sigma_y = ip.gauss_fit(y_proj)
-        self.assertEqual(int(a_x), 30373)
-        self.assertEqual(int(x0_x), 660)
-        self.assertEqual(int(sigma_x), 124)
-        self.assertEqual(int(a_y), 29528)
-        self.assertEqual(int(y0_y), 541)
-        self.assertEqual(int(sigma_y), 127)
+from lcls_tools.common.image_processing.image_processing import ImageProcessor
+from lcls_tools.common.image_processing.roi import RectangularROI
+
+
+class TestImageProcessing(unittest.TestCase):
+    data_location: str = "tests/datasets/images/numpy/"
+
+    def __init__(self, methodName: str = "runTest") -> None:
+        super().__init__(methodName)
+        self.center = [400, 400]
+        self.size = (800, 800)
+        self.widths = (300, 300)
+        self.radius = 50
+        self.image = np.load(self.data_location + "test_roi_image.npy")
+
+    def test_process(self):
+        """
+        Given an np.ndarray and roi process
+        and assert the return in an np.ndarray
+        """
+        image_processor = ImageProcessor()
+        image = image_processor.process(self.image)
+        assert isinstance(image, np.ndarray)
+        roi = RectangularROI(
+            center=self.center, xwidth=self.widths[0], ywidth=self.widths[1]
+        )
+        image_processor = ImageProcessor(roi=roi)
+        image = image_processor.process(self.image)
+        assert isinstance(image, np.ndarray)
```

Comment:
Since ROI is given, it would be good to check the dimensions of the np array match the ROI

You can also use `self.assertIsInstance` and pass a message for when the test fails instead of using raw assert